transport/http.go

```diff
@@ -65,6 +66,9 @@ func NewHTTP(opts HTTPOptions) (Transport, error) {
 	if opts.TargetService == "" {
 		return nil, errMissingTarget
 	}
+	if opts.Method == "" {
+		opts.Method = "POST"
+	}
```

Comment:
Default being applied in two places is a smell. Ideally, either just TransportOptions applies the default or just HTTPOptions. I'm leaning towards the latter.

Reply:
Yeah this definitely isn't ideal, but the goal is:
 * Default to POST in the flags (and have it displayed in the yab help)
 * Match the default if not specified here.

The options are:
 * Only setting it in `NewHTTP`, in which case the CLI help page won't know about it
 * Only set it in `newCommand` or the flag struct, in which case code that does `NewHTTP` without setting a method will use `""`, which defaults to `GET`.

Neither seems great, having 2 different default values is confusing, while not having the default shown in the help page isn't ideal either.

Reply:
Cool, that sounds reasonable.